logic/peer.js

```diff
@@ -117,9 +117,16 @@ class Peer {
 			if (
 				peer[key] !== null &&
 				peer[key] !== undefined &&
-				!_.includes(this.immutable, key)
+				!_.includes(this.required, key)
 			) {
-				this[key] = peer[key];
+				// Update optional httpPort and nonce
+				// for the first time when peer object
+				// has httpPort and nonce as undefined
+				if (!this[key] && _.includes(this.optional, key)) {
+					this[key] = peer[key];
+				} else if (!_.includes(this.optional, key)) {
+					this[key] = peer[key];
+				}
```

Comment:
Suggest to rewrite this logic in a way: 

```
const isOptional = this.optional.includes(key);
const isEmpty = !this[key];

if(!isOptional || (isOptional && isEmpty)) {
   his[key] = peer[key];
}
```
This will make the code more readable.